packages/yoga/src/DropdownInput/web/DropdownInput.jsx

```diff
@@ -0,0 +1,317 @@
+    dropdowninput.containerDropDown.border.color
+  };
+    border-right: ${dropdowninput.containerDropDown.border.width} solid ${
+    dropdowninput.containerDropDown.border.color
+  };
+    border-bottom: ${dropdowninput.containerDropDown.border.width} solid ${
+    dropdowninput.containerDropDown.border.color
+  };
+    list-style: none;
+    padding: 0 ${dropdowninput.containerDropDown.padding.right}px 0 ${
+    dropdowninput.containerDropDown.padding.left
+  }px;
+    transition: all 0.1s ease-in-out;
+    border-radius: 0 0 ${dropdowninput.containerDropDown.border.radius}px ${
+    dropdowninput.containerDropDown.border.radius
+  }px;
+    background: ${dropdowninput.containerDropDown.background};
+    position: absolute;
+    z-index: 9999;
+  `}
+`;
+
+const ButtonDropDown = styled.button`
+  display: flex;
+  align-items: center;
+  width: auto;
+  background: none;
+  border: none;
+  outline: none;
+  cursor: pointer;
+  padding: 0;
+  height: 100%;
+`;
+
+const ItemDropDown = styled.li`
+  ${display}
+  ${flexes}
+  width: 100%;
+  height: 72px;
+  cursor: pointer;
+
+  div {
+    display: flex;
+    align-items: center;
+  }
+`;
+
+const ContainerName = styled.div``;
+
+const DropdownInput = ({
+  full,
+  placeholder,
+  onClear,
+  value,
+  selectedCountry,
+  onChangeSelectedCountry,
+  countries,
+  ...other
+}) => {
+  const [showDropDown, setShowDropDown] = useState(false);
+  const inputRef = useRef(null);
+  const DropDownRef = useRef(null);
+  const containerRef = useRef(null);
+
+  const toggleShowDropDown = () => {
+    setShowDropDown(!showDropDown);
+  };
+
+  useOnClickOutside(DropDownRef, e => {
+    if (e.target.id !== 'button-drop-down') {
+      toggleShowDropDown();
+    }
+  });
+
+  const changeSelectedCountry = selected => () => {
+    onChangeSelectedCountry(selected);
+    toggleShowDropDown();
+    inputRef.current.focus();
+  };
+
+  const renderListCountries = () => {
+    return countries.map(country => {
+      const isCountrySelected = selectedCountry.id === country.id;
+
+      return (
+        <ItemDropDown
+          d="flex"
+          alignItems="center"
+          justifyContent="flex-start"
+          alignContent="center"
+          key={country.id}
+          onClick={changeSelectedCountry(country)}
+        >
+          <ContainerName>
+            <Box>
+              <Icon
+                as={country.icon}
+                width="medium"
+                height="medium"
+                marginRight={4}
+              />
+            </Box>
+            <Text.Small
+              fw={isCountrySelected ? 'medium' : 'regular'}
+              color={isCountrySelected ? 'primary' : 'secondary'}
+            >
+              {country.name}
+            </Text.Small>
+          </ContainerName>
+          {isCountrySelected && (
+            <Icon as={Check} width="medium" height="medium" fill="vibin" />
+          )}
+        </ItemDropDown>
+      );
+    });
+  };
```

Comment:
Creating this `renderListCountries` function is an [anti-pattern](https://youtu.be/b0IZo2Aho9Y?t=117).

It's better to do the `.map` directly and (optionally) extract `<IterDropDown...`  to a new component outside render.

Reply:
dont' forget the solution we've discussed offline

Reply:
✅